File: FlutterUI/flutter_ui/lib/core/services/signalRService.dart
import 'package:flutter_ui/core/utilities/service.dart';
import 'package:flutter_ui/environments/api.dart';
import 'package:signalr_netcore/hub_connection.dart';
import 'package:signalr_netcore/hub_connection_builder.dart';

class SignalRService extends Service {
  HubConnection _hubConnection;

  Future<void> start(String hubUrl) async {
    _hubConnection = HubConnectionBuilder().withUrl(IMAGE_URL + hubUrl).build();
    await _hubConnection.start();
  }

  Future<void> startManually(HubConnection hubConnection) async {
    _hubConnection = hubConnection;
    await _hubConnection.start();
  }

  HubConnection getHubConnection() {
    return _hubConnection;
  }

  Future<Object> invoke(String methodName, List<Object> args) async {
    final result = await _hubConnection.invoke(methodName, args: args);
    return result;
  }

  void on(String methodName, MethodInvocationFunc method) {
    _hubConnection.on(methodName, method);
  }
}

File: AngularUI/UI/src/app/core/services/signal-r.service.ts
import { Injectable } from '@angular/core';
import { HubConnection, HubConnectionBuilder } from '@aspnet/signalr';
import { imageUrl } from 'src/api';

@Injectable({
  providedIn: 'root',
})
export class SignalRService {
  private hubConnection: HubConnection;
  constructor() {}

  start(url: string): void {
    this.hubConnection = new HubConnectionBuilder()
      .withUrl(imageUrl + url)
      .build();
    this.hubConnection.start();
  }

  startManually(hubConnection: HubConnection): void {
    this.hubConnection = hubConnection;
    this.hubConnection.start();
  }

  getHubConnection(): HubConnection {
    return this.hubConnection;
  }

  invoke(methodName: string, args: any[]): Promise<any> {
    return this.hubConnection.invoke(methodName, args);
  }

  on(methodName: string, method: (...args: any[]) => void): void {
    this.hubConnection.on(methodName, method);
  }
}
